### ISSUE TITLE
correct likelihoods

### ISSUE THREAD

**Issue by marcusrosenblatt** [dkaschek/dMod#54]
Hi,

there is another conceptual thing that we came across when comparing likelihood values between the frameworks. It concerns log(or log10)-transformed observables. In this case we have a sigma for the observed data point (sth like obs_a = log(A)) and assume log-normally distribution. In the calculation of the likelihood we add the term log(2 pi sigma^2) to chi2. But for the correct likelihood (to be consistent in the definition), the sigma would have to be rescaled such that one has to add log(2 pi (sigma * exp(datapoint))) for log-transformation and log(2 pi (sigma * 10^(datapoint) * log(10)) for log10-transformation.

This is again just a constant in the objective value. We are thinking of adding it to normL2. But this means to add another column "scale" or "transformation" to the datalist, so we wanted to have your confirmation first.

Best,
Marcus

**Comment by dkaschek** [comment]
I don't agree. When the modeler log-transforms that data it is the modeler's task to also transform sigma and lloq. However, it would make sense to define a function `trans: datalist-> datalist` that does these steps automatically, i.e. transforms sigma according to Gaussian error propagation and applies the transformation to loq.

**Comment by marcusrosenblatt** [comment]
Hi Daniel,

maybe I did not describe the point precisely enough. It is not about transforming sigma in this sense. Of course this is the modelers job. In the example I meant, the chi2 is also computed correctly. So the transformation of the sigma was done correctly. But the calculation of the likelihood is different to what you obtain, if you calculate it analytically. The point is that when you assume log-normal distribution, then the normalization of the likelihood is different.

Best,
Marcus

**Comment by dlill** [comment]
Can you back your claim with some equations? In my intuition, when you assume log-normal distribution, the likelihood looks different from the formula which is implemented in normL2, and the change in the likelihood formula should go beyond a constant.

**Comment by marcusrosenblatt** [comment]
Maybe we take [https://en.wikipedia.org/wiki/Log-normal_distribution] as a start. On the right side you see the normalization of the likelihood I am talking about (PDF). Let us say obs=ln(x), then sigma is the error of obs. In the normalization, there is an additional factor 1/x=1/exp(obs).

@dlill : Exactly, the change in the likelihood is only a constant. So no problem for any modeling/statistical tests. So we have no wrong implementation! But to get the correct likelihood we need this additional factor 1/exp(obs) for all data points corresonding to observables that are assumed tob e log-normally distributed. Since a model might consist of both observations on log-scale and on normal scale, I see no way without adding the information on the transformation to the datalist.

**Comment by dlill** [comment]
i don't really get it, pdfs always need to be transformed *with* their jacobian, i.e. p(x) = p'(x') does not hold. It's p(x)dx = p'(x')dx' that holds. And it's a difference to say that the data is normally distributed around ln(x) on the linear scale or that the log of the data is normally distributed around ln(x).

Maybe a small example with code would help me understand your point :)
